### ISSUE TITLE
SA1024 (ColonsMustBeSpacedCorrectly) doesn't report

### ISSUE THREAD

**Issue by Maxwe11** [DotNetAnalyzers/StyleCopAnalyzers#895]
[SA1024](http://www.stylecop.com/docs/SA1024.html) doesn't report diagnostic on the following code (there is no space before colons):
```C#
using System;

public class Foo<T>: object where T: IFormattable // 2 diagnostics should be here
{
    public Foo(): base() // one here
    {
    }
    public Foo(int x): this() // one here
    {
        Bar(value: x > 2 ? 2: 3); // one here
    }

    private int Bar(int value)
    {
    _label:
        switch (value)
        {
            case 2:
            case 3:
                return value;
            default:
                goto _label;
        }
    }
}
```
